purchase_ext_sst/models/purchase_order.py

```diff
@@ -84,113 +95,96 @@ def open_record(self):
 
     @api.multi
     def write(self, vals):
-        res = super(PurchaseOrder, self).write(vals)
-        val_phone = 'phone' in vals and vals['phone'] or False
-        val_mobile = 'mobile' in vals and vals['mobile'] or False
-        val_tent_name = 'tentative_name' in vals and vals['tentative_name'] \
-                         or False
         for order in self:
-            phone = val_phone or order.phone
-            mobile = val_mobile or order.mobile
-            tent_name = val_tent_name or order.tentative_name
-            if self.is_default_partner(order.partner_id.id):
-                if val_phone or val_mobile:
-                    partner_e = self.get_partner_from_phone(phone, mobile)
-                    if partner_e and order.partner_id != partner_e:
-                        self.set_particulars_from_partner(partner_e, phone,
-                                                          mobile)
-                    if not partner_e:
-                        order.partner_id = self.create_partner(phone, mobile)
-                    self.update_partner(order.partner_id, phone, mobile,
-                                        tent_name)
-            else:
-                if val_phone or val_mobile or val_tent_name:
-                    partner_e = self.get_partner_from_phone(phone,
-                                                            mobile)
-                    if partner_e and order.partner_id != partner_e:
-                        self.update_partner(partner_e, phone, mobile,
-                                            tent_name)
-                        self.set_particulars_from_partner(partner_e, phone,
-                                                          mobile)
-                    else:
-                        self.update_partner(order.partner_id, phone, mobile,
-                                            tent_name)
-
-            if 'order_line' in vals:
-                for line in order.order_line:
-                    product = line.product_id
-                    if product.shop_id != order.shop_id:
-                        product.shop_id = order.shop_id
-                    if product.purchased_by_id != order.purchased_by_id:
-                        product.purchased_by_id = order.purchased_by_id
-        return res
-
-    @api.model
-    def create(self, vals):
-        val_phone = 'phone' in vals and vals['phone'] or False
-        val_mobile = 'mobile' in vals and vals['mobile'] or False
-        val_tent_name = 'tentative_name' in vals and vals['tentative_name'] \
+            val_phone_search = 'phone_search' in vals and vals[
+                'phone_search'] or False
+            val_phone = 'phone_update' in vals and vals['phone_update'] or \
+                        val_phone_search or False
+            val_mobile = 'mobile_update' in vals and vals['mobile_update'] \
                          or False
```

Comment:
I prefer to have these three variables outside the for loop unless you have a reason to keep them inside.  I'm curious what happens in case multiple orders are involved, though.  Haven't checked this part myself.